test/test_selector.py

```diff
@@ -1,18 +1,87 @@
 from test_suites import create_end_to_end_test_suites
+from collections import OrderedDict
+from test_suites import EndToEndTestSuite
+from cloud_platform import CloudPlatform
 import test_suit
 
+
 # TestSelector is responsible for selecting a subset of tests to be run
 # It is meant to filter tests by platform, cloud vendor or any other predicate needed
 class TestSelector:
 
-    def selectTestsToBeRun(self, driver, nameSalt, schemaRegistryAddress, testPlatform, allowedTestsCsv):
-        test_suites = create_end_to_end_test_suites(driver, nameSalt, schemaRegistryAddress, testPlatform, allowedTestsCsv)
+    def select_tests_to_be_run(
+        self,
+        driver,
+        name_salt: str,
+        schema_registry_address: str,
+        test_platform: str,
+        cloud_platform: CloudPlatform,
+        allowed_tests_csv: str,
+    ):
+        test_suites = create_end_to_end_test_suites(
+            driver, name_salt, schema_registry_address, test_platform
+        )
+        test_suites = self.__filter_by_allow_list(allowed_tests_csv, test_suites)
```

Comment:
I am not very skilled with Python so don't hesitate to suggest something more sophisticated here.

Reply:
we could have here just one generic filtering function and pass a predicate lambda but i;m not python expert as well, ask copilot ;p